File: app/src/main/java/com/example/jianming/dao/UpdataStampDao.java
package com.example.jianming.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.jianming.beans.UpdateStamp;

import java.util.List;

@Dao
public interface UpdataStampDao {
    @Query("select * from UpdateStamp where tableName = :tableName")
    UpdateStamp getUpdateStampByTableName(String tableName);

    @Update
    void update(UpdateStamp updateStamp);

    @Insert
    void save(UpdateStamp albumStamp);

    @Delete
    void deleteAll(List<UpdateStamp> updateStampList);
}

File: app/src/main/java/com/example/jianming/dao/UpdataStampDao.kt
package com.example.jianming.dao;


import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.jianming.beans.UpdateStamp;


@Dao
interface UpdataStampDao {
    @Query("select * from UpdateStamp where table_name = :tableName")
    fun getUpdateStampByTableName(tableName: String):UpdateStamp?

    @Update
    fun update(updateStamp: UpdateStamp)

    @Insert
    fun save(sectionStamp: UpdateStamp)

    @Delete
    fun deleteAll(updateStampList: List<UpdateStamp>)
}
